service/storage_service.cc

```diff
@@ -1199,6 +1199,9 @@ std::vector<canonical_mutation> storage_service::build_mutation_from_join_params
     topology_request_tracking_mutation_builder rtbuilder(params.request_id);
     rtbuilder.set("initiating_host",_group0->group0_server().id().uuid())
              .set("done", false);
+    if (_db.local().features().topology_requests_type_column) {
+        rtbuilder.set("request_type", params.replaced_id ? topology_request::replace : topology_request::join);
```

Comment:
`rtbuilder` does not have an access to `features`, so we would still need to make a change on every call to pass it to that. I'd rather leave it as is.

Reply:
We can pass the feature in the constructor of `rtbuilder`, so it can check it internally. Note that we need not pass the entire feature, a bool is enough.